File: pipelines/load_acc_status_bronze/code/job/graph/Source_0.py
from pyspark.sql import *
from pyspark.sql.functions import *
from pyspark.sql.types import *
from job.config.ConfigStore import *
from job.udfs.UDFs import *

def Source_0(spark: SparkSession) -> DataFrame:
    if Config.fabricName == "recipes_fabric":
        return spark.read\
            .schema(
              StructType([
                StructField("acc_id", StringType(), True), StructField("person_id", StringType(), True), StructField("product_id", StringType(), True), StructField("business_date", StringType(), True), StructField("balance", StringType(), True)
            ])
            )\
            .option("header", True)\
            .option("sep", ",")\
            .option("recursiveFileLookup", True)\
            .csv("dbfs:/Prophecy/anshuman@simpledatalabs.com/fin_reporting/external/acc_status/")
    else:
        raise Exception("No valid dataset present to read fabric")

File: pipelines/load_transactions_bronze/code/src/main/scala/graph/Source_0.scala
package graph

import io.prophecy.libs._
import org.apache.spark._
import org.apache.spark.sql._
import org.apache.spark.sql.functions._
import org.apache.spark.sql.types._
import config.ConfigStore._

object Source_0 {

  def apply(spark: SparkSession): DataFrame = {
    Config.fabricName match {
      case "recipes_fabric" =>
        spark.read
          .format("parquet")
          .option("recursiveFileLookup", true)
          .schema(
            StructType(
              Array(
                StructField("acc_id",        StringType, true),
                StructField("tran_id",       StringType, true),
                StructField("business_date", StringType, true),
                StructField("tran_amount",   StringType, true),
                StructField("tran_type",     StringType, true),
                StructField("tran_ts",       StringType, true)
              )
            )
          )
          .load(
            "dbfs:/Prophecy/anshuman@simpledatalabs.com/fin_reporting/external/transactions/"
          )
      case _ =>
        throw new Exception("No valid dataset present to read fabric")
    }
  }

}
